internal/osutil/user.go

```diff
@@ -75,3 +79,48 @@ func UidGid(u *user.User) (sys.UserID, sys.GroupID, error) {
 
 	return sys.UserID(uid), sys.GroupID(gid), nil
 }
+
+// NormalizeUidGid returns the "normalized" UID and GID for the given IDs and
+// names. If both uid and username are specified, the username's UID must match
+// the given uid (similar for gid and group), otherwise an error is returned.
+func NormalizeUidGid(uid, gid *int, username, group string) (*int, *int, error) {
```

Comment:
The API here is a bit atypical/awkward. I'd be worried if this was public, but since it's an "internal/" package, sounds okay and we can tune it later if we ever find the need.